diesel/src/query_builder/upsert/on_conflict_actions.rs

```diff
@@ -90,6 +91,25 @@ where
     }
 }
 
+impl<DB, T>
+    QueryFragment<DB, crate::backend::sql_dialect::on_conflict_clause::MysqlLikeOnConflictClause>
+    for DoUpdate<T>
+where
+    DB: Backend,
+    T: QueryFragment<DB>,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, DB>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+        if self.changeset.is_noop(out.backend())? {
+            out.push_sql(" DO NOTHING");
+        } else {
```

Comment:
Wasn't exactly sure what this was for but opted to just copy pasta in case

Reply:
The `AsChangeset` derives allows to ignore fields that are set to `None`. If all fields are set to `None` the corresponding action is empty. I think in this case we might be able to work around that by getting the corresponding primary key from here: https://github.com/diesel-rs/diesel/blob/983209a6188392552efa51fe76284eb3097b012a/diesel/src/upsert/on_conflict_extension.rs#L388 (likely requires changing `T: QuerySource` to `T: Table`)

Reply:
Sigh. Even without modifying `T: QuerySource` to `T: Table`, I think the tricky thing here is related to the difficulties I'm having with getting `DO NOTHING` to work at all. So I've gotten rid of this `if/else` block and added a TODO for the moment